src/Nethermind/Nethermind.BeaconNode.Peering/SignedBeaconBlockProcessor.cs

```diff
@@ -72,45 +70,57 @@ public class GossipSignedBeaconBlockProcessor : QueueProcessorBase<SignedBeaconB
             }
         }
 
-        protected override async Task ProcessItemAsync(SignedBeaconBlock signedBeaconBlock)
+        public void Enqueue(SignedBeaconBlock signedBeaconBlock, string peerId)
+        {
+            EnqueueItem((signedBeaconBlock, peerId));
+        }
+
+        public void EnqueueGossip(SignedBeaconBlock signedBeaconBlock)
+        {
+            EnqueueItem((signedBeaconBlock, null));
+        }
+
+        protected override async Task ProcessItemAsync((SignedBeaconBlock signedBeaconBlock, string? peerId) item)
         {
             try
             {
                 if (_logger.IsDebug())
-                    LogDebug.ProcessGossipSignedBeaconBlock(_logger, signedBeaconBlock.Message, null);
+                    LogDebug.ProcessSignedBeaconBlock(_logger, item.signedBeaconBlock.Message, item.peerId ?? "gossip",
+                        null);
 
                 if (_mothraConfigurationOptions.CurrentValue.LogSignedBeaconBlockJson)
                 {
                     string logDirectoryPath = GetLogDirectory();
-                    string fileName = string.Format("signedblock{0:0000}_{1}.json",
-                        (int) signedBeaconBlock.Message.Slot,
-                        signedBeaconBlock.Signature.ToString().Substring(0, 10));
+                    string fileName = string.Format("signedblock{0:0000}_{1}{2}.json",
+                        (int) item.signedBeaconBlock.Message.Slot,
+                        item.signedBeaconBlock.Signature.ToString().Substring(0, 10),
+                        item.peerId == null ? "" : "_" + item.peerId);
                     string path = _fileSystem.Path.Combine(logDirectoryPath, fileName);
                     using (Stream fileStream = _fileSystem.File.OpenWrite(path))
                     {
-                        await JsonSerializer.SerializeAsync(fileStream, signedBeaconBlock, _jsonSerializerOptions)
+                        await JsonSerializer.SerializeAsync(fileStream, item.signedBeaconBlock, _jsonSerializerOptions)
```

Comment:
shall we use IJsonSerializer and EthereumJsonSerializer from Eth1?

Reply:
Is that using the new core 3 JSON serializer (UTF8) or JSON.NET?   I don't think I need anything more beyond what core 3 has for now (and these are just debug output of blocks received).  When I fix the API (soon, I hope), I'm planning to drop nswag and just implement directly, and that natively uses the core 3 serializer, and I will look at the Eth1 library for that.

Reply:
Won't stop merge; look at later as per above.